tests/test_abs.py

```diff
@@ -0,0 +1,185 @@
+# -------------------------------------------------------------------------
```

Comment:
Could you put this in the data subfolder and give it a long descriptive name? Not as if we need to type it. The pattern I use is test_name_of_module_being_tested.py - makes it easier to work out what is doing what.